pkg/restapi/vc/operation/operations.go

```diff
@@ -689,11 +689,27 @@ func (o *Operation) retrieveCredentialHandler(rw http.ResponseWriter, req *http.
 		return
 	}
 
-	docURLs, err := o.queryVault(profile, id)
-	if err != nil {
-		o.writeErrorResponse(rw, http.StatusInternalServerError, err.Error())
+	var docURLs []string
 
-		return
+	for retries := 0; retries < 10; retries++ {
+		if retries != 0 {
+			time.Sleep(time.Second * 5)
```

Comment:
5 seconds is too long in between first and second retry

Reply:
@bstasyszyn

Reply:
also 50 seconds maximum is too much :).

Reply:
For example, see: https://github.com/trustbloc/fabric-mod/blob/ba9a41234a5002930b1232c4640b86fbd1bf9bb7/common/util/retry/retry.go#L47

Reply:
Reworked it - take a look and let me know.